actionpack/lib/action_dispatch/middleware/cookies.rb

```diff
@@ -563,24 +566,31 @@ def serializer
         def digest
           request.cookies_digest || "SHA1"
         end
-
-        def key_generator
-          request.key_generator
-        end
     end
 
-    class SignedCookieJar < AbstractCookieJar # :nodoc:
+    class SignedKeyRotatingCookieJar < AbstractCookieJar # :nodoc:
       include SerializedCookieJars
 
       def initialize(parent_jar)
         super
-        secret = key_generator.generate_key(request.signed_cookie_salt)
-        @verifier = ActiveSupport::MessageVerifier.new(secret, digest: digest, serializer: ActiveSupport::MessageEncryptor::NullSerializer)
+
+        secret = key_generator.generate_key(signed_cookie_salt)
+        @verifier = ActiveSupport::MessageVerifier.new(secret, digest: signed_cookie_digest, serializer: SERIALIZER)
+
+        Array.wrap(signed_cookie_rotate_out).each do |rotation_options|
```

Comment:
With the proposed `config.action_dispatch.rotations.cookies` this would probably read like `request.cookie_rotations.signed.each do …` and likewise `request.cookie_rotations.encrypted.each do …` (don't think we'd even need the `wrap` then.

Reply:
Going to get things integrated using this new configuration API. Will take a little bit of time cause I need to update some tests then 😃